### ISSUE TITLE
error by execute sat.sh

### ISSUE THREAD

**Issue by smarti2019** [stdevPavelmc/FAASGS#11]
error :

root@rasp4b:/home/pi/FAASGS# ./sat.sh
Updating data and scheduling new sats
Fresh data fetched, updating the good file
Filtering the sats we need and set simple names
=> NOAA 19/NOAA19
./sat.sh: line 89: [: file: integer expression expected
=> NOAA 18/NOAA18
./sat.sh: line 89: [: file: integer expression expected
=> NOAA 15/NOAA15
./sat.sh: line 89: [: file: integer expression expected

any ideas?

**Comment by stdevPavelmc** [comment]
Please upgrade the the latest changes and try again. Do give feedback if the error persist.

See README.md about how to upgrade

**Comment by smarti2019** [comment]
yes, the same error 
./sat.sh: line 89: [: file: integer expression expected

**Comment by stdevPavelmc** [comment]
Hum... let me check that...

```if [ $MAXELEV -gt 10 ] ; then```

So your MAXELEV var does not contains an integer

Please run this on the PC you have the soft installed and report the full output:

```sh
/usr/bin/predict -t /etc/sat_data/data.txt -p NOAA19 
```

**Comment by Murzim2** [comment]
Try this it fixed the problem with predict and it now runs correctly. Now to recheck the rest of the install as still not working
mkdir ~/.predict
cd ~/.predict
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.db
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.tle
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.qth

**Comment by Murzim2** [comment]
Having fixed predict now get this running the command line bit.
pi@raspberrypi:~ $ /usr/bin/predict -t /etc/sat_data/data.txt -p NOAA19
1612437023 Thu 04Feb21 11:10:23    0    2  221   80  356   3421  61814 * 0.000000
1612437124 Thu 04Feb21 11:12:04    2  349  225   77   24   3229  61814 * 0.000000
1612437225 Thu 04Feb21 11:13:45    2  335  229   73   39   3181  61814 * 0.000000
1612437326 Thu 04Feb21 11:15:26    1  322  233   68   48   3285  61814 * 0.000000
1612437386 Thu 04Feb21 11:16:26    0  315  236   64   52   3412  61814 * 0.000000

but still get this when running sats.sh
pi@raspberrypi:~ $ sudo sats.sh
Updating data and scheduling new sats
Fresh data fetched, updating the good file
Filtering the sats we need and set simple names
=> ISS (ZARYA)/ISS
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> RADFXSAT (FOX-1B)/AO91-FOX-1B
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> FOX-1CLIFF (AO-95)/AO95-FOX-1CLIFF
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> FOX-1D (AO-92)/AO92-FOX-1D
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> DIWATA-2B/PO101-DIWATA-2B
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> NOAA 19/NOAA19
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> NOAA 18/NOAA18
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected
=> NOAA 15/NOAA15
/usr/local/bin/sats.sh: line 89: [: file: integer expression expected

**Comment by smarti2019** [comment]
from the console at user "pi"
/usr/bin/predict -t /etc/sat_data/data.txt -p NOAA19
1612454507 Thu 04Feb21 16:01:47    0  101  184   41  330   3419  61817 * 0.000000
1612454608 Thu 04Feb21 16:03:28    5   92  189   46  333   2906  61817 * 0.000000
1612454709 Thu 04Feb21 16:05:09   10   79  193   52  335   2484  61817 * 0.000000
1612454809 Thu 04Feb21 16:06:49   14   62  197   58  339   2211  61817 * 0.000000
1612454908 Thu 04Feb21 16:08:28   15   42  201   63  343   2143  61817 * 0.000000
1612455006 Thu 04Feb21 16:10:06   13   23  205   69  349   2293  61817 * 0.000000
1612455106 Thu 04Feb21 16:11:46    8    8  209   74  358   2624  61817 * 0.000000
1612455206 Thu 04Feb21 16:13:26    3  357  214   78   15   3084  61817 * 0.000000
1612455271 Thu 04Feb21 16:14:31    0  351  216   80   33   3424  61817 * 0.000000

**Comment by stdevPavelmc** [comment]
Please download the attached file into the machine you has FAASGS installed, then do the following:

```sh
mv sats.txt sats.sh
chmod +x sats.sh
./sats.sh
```

And report back the full output (copy it as a txt file and attach it to your post), this file is a debug loaded version of sat.sh

File: [sats.txt](https://github.com/stdevPavelmc/FAASGS/files/5930143/sats.txt)

**Comment by Murzim2** [comment]
Hi Pavel

Here is the output from your debug script.

Dave

**Comment by smarti2019** [comment]
here the output:

pi@rasp4b:~/FAASGS $ ./sats.sh
Updating data and scheduling new sats
mkdir: cannot create directory ‘/var/www’: Permission denied
/etc/sat_data/data_temp.txt: Permission denied
Can't fetch fresh TLE data, using old one
./sats.sh: line 30: /var/www/html/sat/passes.txt: Permission denied
=> ISS (ZARYA)/ISS
===== DEBUG: PREDICTION ======
- /tmp/tmp.Q4a4Jol0Iw -
1612541910 Fri 05Feb21 16:18:30    0  279   76   52   25   2370  26821 * 0.000000
1612541982 Fri 05Feb21 16:19:42    5  276   79   52   18   1890  26821 * 0.000000
1612542053 Fri 05Feb21 16:20:53   11  271   82   52   11   1426  26821 * 0.000000
1612542123 Fri 05Feb21 16:22:03   21  260   85   51    4   1005  26821 * 0.000000
1612542190 Fri 05Feb21 16:23:10   35  233   88   50  358    706  26821 * 0.000000
1612542250 Fri 05Feb21 16:24:10   38  187   91   49  353    654  26821 * 0.000000
1612542306 Fri 05Feb21 16:25:06   28  154   94   47  348    833  26821 * 0.000000
1612542370 Fri 05Feb21 16:26:10   16  138   97   46  343   1176  26821 * 0.000000
1612542439 Fri 05Feb21 16:27:19    8  130  100   43  338   1608  26821 * 0.000000
1612542509 Fri 05Feb21 16:28:29    3  126  103   41  333   2076  26821 * 0.000000
1612542553 Fri 05Feb21 16:29:13    0  124  105   39  330   2369  26821 * 0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.Q4a4Jol0Iw -
===============================
   Fri Feb 05 17:18:30 2021, 38o, 643s
./sats.sh: line 105: /var/www/html/sat/passes.txt: Permission denied
=> RADFXSAT (FOX-1B)/AO91-FOX-1B
===== DEBUG: PREDICTION ======
- /tmp/tmp.P0oGO88CMm -
1612558557 Fri 05Feb21 20:55:57    0   44   72   65  307   3025  17367   0.000000
1612558647 Fri 05Feb21 20:57:27    2   56   76   60  311   2799  17367   0.000000
1612558738 Fri 05Feb21 20:58:58    4   70   80   55  314   2692  17367   0.000000
1612558830 Fri 05Feb21 21:00:30    4   85   84   49  316   2723  17367   0.000000
1612558924 Fri 05Feb21 21:02:04    3   99   88   44  318   2891  17367   0.000000
1612559018 Fri 05Feb21 21:03:38    0  111   92   38  320   3175  17367   0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.P0oGO88CMm -
===============================
=> FOX-1CLIFF (AO-95)/AO95-FOX-1CLIFF
===== DEBUG: PREDICTION ======
- /tmp/tmp.99kArZfnqi -
1612549066 Fri 05Feb21 18:17:46    0   71  200   54  312   2803  11877   0.000000
1612549150 Fri 05Feb21 18:19:10    2   58  204   59  314   2573  11877   0.000000
1612549234 Fri 05Feb21 18:20:34    3   44  207   64  318   2473  11877   0.000000
1612549318 Fri 05Feb21 18:21:58    3   29  211   69  323   2518  11877   0.000000
1612549401 Fri 05Feb21 18:23:21    1   15  215   74  330   2702  11877   0.000000
1612549434 Fri 05Feb21 18:23:54    0   11  216   76  334   2807  11877 + 0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.99kArZfnqi -
===============================
=> FOX-1D (AO-92)/AO92-FOX-1D
===== DEBUG: PREDICTION ======
- /tmp/tmp.hr53QdUB6F -
1612551000 Fri 05Feb21 18:50:00    0   79   63   51  316   2573  17058   0.000000
1612551078 Fri 05Feb21 18:51:18    2   67   66   56  318   2327  17058   0.000000
1612551155 Fri 05Feb21 18:52:35    4   52   70   61  321   2201  17058   0.000000
1612551232 Fri 05Feb21 18:53:52    4   37   73   66  324   2217  17058   0.000000
1612551310 Fri 05Feb21 18:55:10    2   22   77   70  329   2374  17058   0.000000
1612551388 Fri 05Feb21 18:56:28    0   11   80   75  337   2647  17058   0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.hr53QdUB6F -
===============================
=> DIWATA-2B/PO101-DIWATA-2B
===== DEBUG: PREDICTION ======
- /tmp/tmp.yHyFvbS3yW -
1612562981 Fri 05Feb21 22:09:41    0   90  204   47  316   2825  12382   0.000000
1612563066 Fri 05Feb21 22:11:06    3   78  207   52  318   2471  12382   0.000000
1612563150 Fri 05Feb21 22:12:30    6   64  211   57  320   2230  12382   0.000000
1612563235 Fri 05Feb21 22:13:55    7   47  215   62  323   2142  12382   0.000000
1612563319 Fri 05Feb21 22:15:19    6   30  218   67  328   2224  12382   0.000000
1612563403 Fri 05Feb21 22:16:43    4   16  222   72  334   2461  12382   0.000000
1612563487 Fri 05Feb21 22:18:07    0    4  226   77  345   2812  12382   0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.yHyFvbS3yW -
===============================
=> NOAA 19/NOAA19
===== DEBUG: PREDICTION ======
- /tmp/tmp.mxxwiNhLnn -
1612546044 Fri 05Feb21 17:27:24    0  173  175   25  348   3412  61832 * 0.000000
1612546145 Fri 05Feb21 17:29:05    7  175  179   31  350   2752  61832 * 0.000000
1612546246 Fri 05Feb21 17:30:46   16  179  184   37  352   2111  61832 * 0.000000
1612546344 Fri 05Feb21 17:32:24   29  186  188   42  354   1533  61832 * 0.000000
1612546434 Fri 05Feb21 17:33:54   47  204  191   48  356   1114  61832 * 0.000000
1612546504 Fri 05Feb21 17:35:04   61  247  194   51  357    962  61832 * 0.000000
1612546554 Fri 05Feb21 17:35:54   57  290  196   54  359    998  61832 * 0.000000
1612546610 Fri 05Feb21 17:36:50   44  315  199   58    1   1172  61832 * 0.000000
1612546685 Fri 05Feb21 17:38:05   29  328  202   62    4   1532  61832 * 0.000000
1612546774 Fri 05Feb21 17:39:34   17  335  206   67    9   2058  61832 + 0.000000
1612546872 Fri 05Feb21 17:41:12    8  339  210   72   16   2675  61832 + 0.000000
1612546973 Fri 05Feb21 17:42:53    1  342  214   76   30   3329  61832 + 0.000000
1612546987 Fri 05Feb21 17:43:07    0  342  215   77   32   3424  61832 + 0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.mxxwiNhLnn -
===============================
   Fri Feb 05 18:27:24 2021, 61o, 943s
./sats.sh: line 105: /var/www/html/sat/passes.txt: Permission denied
=> NOAA 18/NOAA18
===== DEBUG: PREDICTION ======
- /tmp/tmp.VOO4kjAHe8 -
1612544656 Fri 05Feb21 17:04:16    0   69   85   54  304   3431  80993   0.000000
1612544758 Fri 05Feb21 17:05:58    3   57   89   60  307   3091  80993   0.000000
1612544860 Fri 05Feb21 17:07:40    6   43   94   65  312   2882  80993 * 0.000000
1612544961 Fri 05Feb21 17:09:21    6   28   98   71  319   2837  80993 * 0.000000
1612545063 Fri 05Feb21 17:11:03    5   13  102   76  331   2965  80993 * 0.000000
1612545164 Fri 05Feb21 17:12:44    2  360  106   80  352   3243  80993 * 0.000000
1612545222 Fri 05Feb21 17:13:42    0  354  109   81   11   3453  80993 * 0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.VOO4kjAHe8 -
===============================
=> NOAA 15/NOAA15
===== DEBUG: PREDICTION ======
- /tmp/tmp.l2xcBH601u -
1612541462 Fri 05Feb21 16:11:02    0  111   64   37  319   3320  18246 * 0.000000
1612541560 Fri 05Feb21 16:12:40    5  103   68   43  321   2780  18246 * 0.000000
1612541659 Fri 05Feb21 16:14:19   11   91   72   49  323   2315  18246 * 0.000000
1612541756 Fri 05Feb21 16:15:56   16   74   76   54  326   1986  18246 * 0.000000
1612541851 Fri 05Feb21 16:17:31   19   52   80   60  329   1863  18246 * 0.000000
1612541945 Fri 05Feb21 16:19:05   17   30   84   65  334   1974  18246 * 0.000000
1612542040 Fri 05Feb21 16:20:40   12   13   88   70  340   2287  18246 * 0.000000
1612542138 Fri 05Feb21 16:22:18    6    1   93   75  351   2741  18246 * 0.000000
1612542236 Fri 05Feb21 16:23:56    1  353   97   79   10   3280  18246 * 0.000000
1612542248 Fri 05Feb21 16:24:08    0  352   97   80   13   3348  18246 * 0.000000
===============================
===== DEBUG: MAXELEV ======
- /tmp/tmp.l2xcBH601u -
===============================
   Fri Feb 05 17:11:02 2021, 19o, 786s
./sats.sh: line 105: /var/www/html/sat/passes.txt: Permission denied
pi@rasp4b:~/FAASGS $

**Comment by stdevPavelmc** [comment]
Ouch!

Sorry, I must said: "run it as sudo", that's why it give you a lot of "Permission denied"

So please run it as `sudo ./sats.sh`  and report back.

**Comment by Murzim2** [comment]
Hi Pavel

Think I may have seen another problem associated with Predict. The .Predict directory was not in /root
Now get entry's in the web page.
Will see how I get on tomorrow.

Attached the output from the system with the /root/.predict in place.

Dave, G8PGO

**Comment by stdevPavelmc** [comment]
Ok, last run (debug script) is looking good, please notify me if the issue is fixed to close it.

**Comment by Murzim2** [comment]
Hi Pavel

Yes I think that issue is fixed by doing the following as there were errors in the Predict install.


  1.  mkdir ~/.predict
cd ~/.predict
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.db
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.tle
wget https://raw.githubusercontent.com/kd2bd/predict/master/default/predict.qth
  2.  sudo -i

ln -s /home/pi/.predict .predict

create a symbolic link from /root to /home/pi/.predict



Think you can now close this one.

Now re-looking at the rest of the install to see if I have done it correctly as I am getting

The web pages up to the recorded Satellite data but if I click the play button of the image icon I get

A 404 not found error for both the wav and png files.

**Comment by stdevPavelmc** [comment]
> Yes I think that issue is fixed by doing the following as there were errors in the Predict install.

Good to know!

**Comment by stdevPavelmc** [comment]
> The web pages up to the recorded Satellite data but if I click the play button of the image icon I get A 404 not found error for both the wav and png files.

Let it record a few passes, some times when two satellites pass overlaps just one of them will be recorded and the other just got empty.

I wait your feedback.

**Comment by stdevPavelmc** [comment]
Hum... O think this maybe related with https://github.com/stdevPavelmc/FAASGS/issues/13 

TL;DR: if you use RaspiOS then `rtl_fm` has no '-E wav' option.

I will test it today and will try to came up with a solution for all
